var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -173,6 +174,49 @@ def write_empty_libs(f, provider):
                     f.write('libraries=%s\n' % lapackblas_names)
                     write_library_dirs(f, lapackblas_dirs)
 
+        # See: https://github.com/numpy/numpy/pull/12991
+        # Bug introduced: https://github.com/numpy/numpy/pull/10665
+        # So affects 1.15.* => 1.16.* releases
+        # Patch not needed after numpy@1.17.0
+        if '%intel' in spec and spec.satisfies('@1.15.0:1.16.999'):
+            # Need to add #pragma intel optimization_level 2, cannot use
+            # llnl.util.filter_file because this is a multiline match, but a
```

Comment:
Note that in general you should be able to do regular expressions with newlines. `re.MULTILINE` can be used for a more-intuitive handling of `^` and `$`, but I don't see that in your expression - sorry if I missed it. As a sanity check, does the following fail:

```
filter_file(
  r"(NPY_NO_EXPORT void\n"
  r"@TYPE@_@kind@\(char \*\*args, npy_intp \*dimensions, npy_intp \*steps, void \*NPY_UNUSED\(func\)\))",  # noqa: E501
  # Add #pragma patch and paste back function signature
  "#ifdef __INTEL_COMPILER\n"
  "    #pragma intel optimization_level 2\n"
  "#endif\n"
  r"\1",
  filename
)
```

See also:

* https://docs.python.org/2/library/re.html
* https://stackoverflow.com/questions/14689531/how-to-match-a-new-line-character-in-python-raw-string